File: Python/14-链表中倒数第k个结点.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def FindKthToTail(self, head, k):
        # write code here
        n = 0;
        p = head;
        while p:
            n += 1;
            p = p.next;
        if k > n:
            return None
        p = head;
        for i in range(n - k):
            p = p.next
        return p

File: C++/14-链表中倒数第k个结点.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
        ListNode* p = pListHead;
        int n = 0;
        while(p){
            ++n;
            p = p->next;
        }
        p = pListHead;
        if(k > n)
            return 0;
        for(int i = 0; i < n - k; ++i)
            p = p->next;
        return p;
    }
};
